src/tests/ftest/pool/list_verbose.py

```diff
@@ -0,0 +1,332 @@
+            },
+            {
+                "tier_name": "NVME",
+                "size": nvme_size,
+                "free": nvme_free,
+                "imbalance": nvme_imbalance
+            }
+            ]
+        }
+
+    @staticmethod
+    def get_scm_nvme_free(pool_list_out):
+        """Get SCM and NVMe free values.
+
+        Args:
+            pool_list_out (dict): pool list output dictionary to obtain the free
+                values from.
+
+        Returns:
+            tuple: SCM free and actual NVMe free values.
+        """
+        scm_free = -1
+        nvme_free = -1
+        for usage in pool_list_out["usage"]:
+            if usage["tier_name"] == "SCM":
+                scm_free = usage["free"]
+            elif usage["tier_name"] == "NVME":
+                nvme_free = usage["free"]
+
+        return (scm_free, nvme_free)
+
+    def verify_two_pools(self, targets_disabled_1=0, scm_size_1=None,
+                         nvme_size_1=None):
+        """Verify two pools by comparing expected and actual dictionaries.
+
+        Args:
+            targets_disabled_1 (int, optional): Targets disabled for pool 1.
+                Defaults to 0.
+            scm_size_1 (int, optional): SCM size for pool 1. Defaults to None.
+            nvme_size_1 (int, optional): NVMe size for pool 1. Defaults to None.
+        """
+        actual_pools = self.pool[0].dmg.get_pool_list_all(verbose=True)
+
+        # Get the dict of 1 and 2. Assume that the first two are what we want.
+        if actual_pools[0]["uuid"] == self.pool[0].uuid.lower():
+            actual_pool_1 = actual_pools[0]
+            actual_pool_2 = actual_pools[1]
+        else:
+            actual_pool_1 = actual_pools[1]
+            actual_pool_2 = actual_pools[0]
+
+        # Get the free values and create expected for pool 1.
+        actual_scm_free_1, actual_nvme_free_1 = self.get_scm_nvme_free(
+            actual_pool_1)
+
+        expected_pool_1 = self.create_expected(
+            pool=self.pool[0], scm_free=actual_scm_free_1,
+            nvme_free=actual_nvme_free_1, targets_disabled=targets_disabled_1,
+            scm_size=scm_size_1, nvme_size=nvme_size_1)
+
+        # Compare expected and actual data for pool 1.
+        self.assertDictEqual(expected_pool_1, actual_pool_1)
+
+        # Get the free values for pool 2.
+        actual_scm_free_2, actual_nvme_free_2 = self.get_scm_nvme_free(
+            actual_pool_2)
+
+        # Pool 2 uses two ranks, so need to double the value in TestPool.
+        scm_size_2 = self.pool[1].scm_size.value * 2
+        nvme_size_2 = self.pool[1].nvme_size.value * 2
+
+        expected_pool_2 = self.create_expected(
+            pool=self.pool[1], scm_free=actual_scm_free_2,
+            nvme_free=actual_nvme_free_2, scm_size=scm_size_2,
+            nvme_size=nvme_size_2, targets_total=16)
+
+        # Compare expected and actual data for pool 2.
+        self.assertDictEqual(expected_pool_2, actual_pool_2)
```

Comment:
Instead of calling `get_scm_nvme_free` multiple times, you could have it gather all the free pool data indexed by pool uuid:
```suggestion
        # Get the free values and create expected for pool 1.
        free_data = self.get_scm_nvme_free(actual_pools)

        expected_pools = []
        for index, pool in enumerate(self.pool):
            expected_pools.append(
                self.create_expected(
                     pool=pool,
                     scm_free=free_data[pool.uuid.lower()]["scm"],
                     nvme_free=free_data[pool.uuid.lower()]["nvme"],
                     targets_disabled=targets_disabled[index],
                     scm_size=scm_size[index],
                     nvme_size=nvme_size[index])
                     
       self.assertListEqual(sorted(expected_pools, key="uuid"), sorted(actual_pools, key="uuid"))
```